server/routes/events.js

```diff
@@ -12,120 +13,444 @@ router.get('/ping', function(req, res, next) {
 
 // GET all events
 router.get('/', function(req, res, next) {
-  db.query('SELECT * FROM event', [], (err, result) => {
-    if (err) return next(err);
-    res.send(result.rows);
-  });
+  knex('event').select()
+    .then(result => {
+      if (result.length) {
+        res.json(result);
+      } else {
+        res.status(404).json('No events found.');
+      }
+    })
+    .catch(err => { return next(err) });
 });
 
 // GET all event names
 router.get('/names', function(req, res, next) {
-  db.query('SELECT fb_id, name FROM event', [], (err, result) => {
-    if (err) return next(err);
-    res.send(result.rows);
-  });
+  knex('event').select('fb_id, name')
+    .then(result => {
+      if(result.length) {
+        res.json(result);
+      } else {
+        res.status(404).json('No event names found.');
+      }
+    })
+    .catch(err => { return next(err) });
 });
 
 // GET all unique event locations
 router.get('/locations', function(req, res, next) {
-  db.query('SELECT DISTINCT location_id FROM event', [], (err, result) => {
-  	if (err) return next(err);
-    res.send(result.rows);
-  });
+  knex('event').select('location_id')
+    .then(result => {
+      if(result.length) {
+        res.json(result);
+      } else {
+        res.status(404).json('No event locations found.')
+      }
+    })
+    .catch(err => { return next(err) });
 });
 
 // GET event by event_id
 router.get('/:event_id/id', function(req, res, next) {
+  knex('event').select().where({ fb_id: req.params.event_id })
+    .then(result => {
+      if(result.length) {
+        res.json(result);
+      } else {
+        res.status(404).json('No events with matching event ID found.');
+      }
+    })
+    .catch(err => { return next(err) });
+});
+
+// Add a single event
+router.post('/', function(req, res, next) {
+  let values = { 
+  	fb_id: req.body.event_id, 
+  	name: req.body.name, 
+  	starts_at: req.body.starts_at, 
+  	ends_at: req.body.ends_at,
+  	quarter: req.body.quarter,
+  	location_id: req.body.location_id,
+  	description: req.body.description,
+  	fb_event: req.body.fb_event,
+  	picture: req.body.picture,
+  	is_featured: req.body.is_featured
+  };
+  
+  if (!req.body.event_id) {
+  	util.throwError(400, "Missing 'event_id' parameter.");
+  }
+  if (!req.body.name) {
+  	util.throwError(400, "Missing 'name' parameter.");
+  }
+  if (!req.body.starts_at) {
+  	util.throwError(400, "Missing 'starts_at' parameter.");
+  }
+  if (!req.body.ends_at) {
+  	util.throwError(400, "Missing 'ends_at' parameter.");
+  }
+  if (!req.body.quarter) {
+  	util.throwError(400, "Missing 'quarter' parameter.");
+  }
+
+  let category_ids = req.body.categories;
+  let category_values = [];
+  let company_ids = req.body.companies;
```

Comment:
What happens if multiple values are not passed to these params? is it still an array? 

If necessary, use Array.isArray() to check if it is an array?